src/meta/src/barrier/mod.rs

```diff
@@ -874,21 +824,16 @@ impl GlobalBarrierManager {
                 };
 
                 for command in finished_commands {
-                    self.checkpoint_control.stash_command_to_finish(command);
+                    tracker.stash_command_to_finish(command);
                 }
 
-                if let Some(command) = cancelled_command {
-                    self.checkpoint_control.cancel_command(command);
-                } else if let Some(table_id) = node.command_ctx.table_to_cancel() {
+                if let Some(table_id) = node.command_ctx.table_to_cancel() {
                     // the cancelled command is possibly stashed in `finished_commands` and waiting
                     // for checkpoint, we should also clear it.
-                    self.checkpoint_control.cancel_stashed_command(table_id);
```

Comment:
Hmm what about this part. The `finished_commands` could still contain the command to be cancelled.

Reply:
The original `cancel_stashed_command` only modifies the `finished_jobs`, which is moved to the tracker in this PR. The original logic of `cancel_stashed_command` is merged with the original `find_cancelled_command` into `cancel_command`.